documentation/4_Perspectives_of_the_Reference_Architecture_Model/4_1_Security_Perspective/4_1_2_Identity_and_Trust_Management.md

```diff
@@ -0,0 +1,113 @@
+In the IDS, each connector instance possesses it's own identity. Each connector instance is made up of several aspects:
+* The platform the IDS Connector instance depends on. A platform consists of hardware, firmware, operating system and (container) runtime environment.
+* The Connector Core Services software artefacts that provide management functionality and IDS interoperability.
+* The configuration of a IDS Connector (defined data routes, configured Usage Control framework).
+* The Data Apps or other services (e.g., Clearing House services) that are bound to this connector instance.
+
+The IDS Connector identity serves to uniquely identify one instance of a service and app bundle on qualified platforms. The identity concept is equally used for all technical components in the IDS.
+
+The service instance for a connector is typically a Service including core services, some usage control framework and applications (Data Apps).
+Other components (Broker, DAPS, ...) are represented by their Service (represented by one or multiple containers) running on a comparable platform.
+
+One component always is characterized by the combination of platform and service instances. As an example, this Connector instance is running several data apps. The identity is comprised of the platform, the Connector Core Services and the deployed Data Apps.
+
+![Components SW Stack](./media/SW_Stack_Components_connector_blueprint.png)
+#### _Figure 4.1.2.1: Components of the Software Stack of an IDS Connector_
+
+### Component Identifier
+
+The identity of a combination of platform and service instance is bound to an identifier for the service instance.
+
+* Each component gets a unique identified (C_UID) bound to the service instance.
+* The uniqueness of this identifier is ensured by the Identity Provider.
+* Each UID is mapped to a Connector Instance Key (CIK) pair which is typically used for TLS but possibly also data signing and other identity proofs.
+
+Each Service Instance needs to be mapped to one platform it utilizes:
+* Each platform blueprint gets a unique identifier during the component certification.
+* For a component with Trust level 1, the concrete platform instantiation (running version of this blueprint) does NOT get a UID as well as key and certificate for this platform instance. Instead the connector description solely references the (unique) identifier of the certified blueprint (and the operator needs to be trusted to ensure its correct instantiation).
+* For components with Trust Level 2 or 3, each platform instantiation needs to be uniquely identified with a UID (P_UID). This UID is required to provide a mapping from service instance to platform.
+* Each P_UID is subject of a certificate for the utilized platform key(s) (e.g. AK of a TPM) which enable verification of the platform integrity.
+* One P_UID can map to Platform Instance Keys (PIK) for 1-n physical devices/protected VMs:
```

Comment:
So every time the exact same deployment is moved from one VM to another - for whatever (maybe technical) reasons, and probably this will happen not that often, anyways - the whole certification process has to be started again?

Reply:
First of all, the certification (as in the IDS certification process for a connector) is only conducted once for each connector SW stack (we called it a blueprint). This blueprint can then be instantiated on an arbitrary number of devices or servers as long as it matches the evaluated and tested SW stack.

However, I think with "certification process" you meant the requesting of an identity certificate for the platform, right?
Here we are expecting cases where one platform does not need to be limited to one device, e.g. when using a distributed Kubernetes setup utilizing multiple servers or VMs as worker nodes which will together have one P_UID. However, each of the worker nodes (server or VM) requires their own key material (PIKs) e.g. for remote attestation, so there needs to be an identity certificate for each of those PIKs mapping them to the P_UID of the platform this worker node belongs to (e.g. "Fraunhofer ISST Kubernetes Cluster 1"). So yes, for each worker node that is added to the cluster, the operator needs to request a new Platform Identity Certificate from the CA. But this certificate can be used as long as the VM or server belongs to this Kubernetes cluster and is not affected by moving Connector Core Services (which I assume you mean with "deployment") from one worker node to another.
Does this clarify your question? 
*

Reply:
```So yes, for each worker node that is added to the cluster, the operator needs to request a new Platform Identity Certificate from the CA```
In a real life this is not achievable when you use a cloud provider. It is not recognizable for a cloud customer which HW server is used for the deployment. This works just in small own-controlled environments. Do you have a recommendation how to cope with this issue in a setting e.g. GCP, AWS or azure?

Reply:
From my perspective: A connector with trust level 1 also works without problems in a cloud environment since you don't need a platform identity certificate for this. In case of trust level 2 or 3, you would need to either gain support from the cloud provider to provide such platform identity certificates and get their platforms certified or you will need to use confidential computing approaches to exclude the components under control of the cloud provider.